scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -463,16 +763,24 @@ function Test-AgentRegistryArgs {
 }
 
 function Get-ContainerOs {
-    if ((Test-Path $MobyInstallDirectory) -or (Test-Path $MobyDataRootDirectory)) {
-        return 'Windows'
+    $yamlPath = (Join-Path -Path $EdgeDataDirectory -ChildPath "config.yaml")
+    if (-not (Test-Path $yamlPath)) {
+        return "Windows"
     }
-    else {
-        return 'Linux'
+    $configurationYaml = Get-Content $yamlPath -Raw
+    if (-not ($configurationYaml -match "moby_runtime:\s*uri:\s*'([^']+)'")) {
```

Comment:
This is brittle - the string may use different quotes or not use quotes at all, there may be other fields before `uri`, etc. Functions like `Set-GatewayAddress` and `Set-ProvisioningMode` only need to parse the *default* config.yaml that shipped with iotedged, so its regexes could make assumptions. You can't make that assumption here since this will run on a user-provided config.

Reply:
I guess the only impact of mis-detecting Linux as Windows is that it'll be unable to delete the Linux Docker containers, right? If it's just that it's not too bad, I suppose.

Reply:
Right. I am all ears if you have better ideas to detect Linux containers. Imho we should have a value in the config for that, but I was not sure if we want to complicate the config file for that.

Reply:
This is okay for now.